src/main/java/bio/terra/cli/service/WorkspaceManagerService.java

```diff
@@ -1148,7 +1148,13 @@ private void handleClientExceptions(
     } catch (ApiException | InterruptedException ex) {
       // if this is a WSM client exception, check for a message in the response body
       if (ex instanceof ApiException) {
-        errorMsg += ": " + logErrorMessage((ApiException) ex);
+        String exceptionErrorMessage = logErrorMessage((ApiException) ex);
+        if (exceptionErrorMessage.contains("User is unauthorized to link spend profile")) {
```

Comment:
I'm a little uneasy about string matching on error messages, a change to WSM might accidentally break this without anyone knowing. Would it be possible to match on a smaller phrase (maybe just "spend profile") and check error code (in this case 403) instead?

Reply:
Good idea. I'll add a check for the status code. Also added a status code check to the place where we check against a SAM error message string.

I agree in general that matching on error messages is error-prone, but not sure a better approach. Maybe in the future we could match on a specific `Exception` class in the `ApiException`'s `cause` field?